### PR TITLE
[iOS] Fix swipe back after resetTo

### PR DESCRIPTION
**Pull request by Rudolph-Miller**
Reproducible mini app is below.
After `Push` to push next screen, `Reset` button to call `resetTo` and `Push` to push next screen, then swipe to back function is now disabled.

It because `viewWillDisappear` will not be called via `setViewControllers`.

```
import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { Navigation } from 'react-native-navigation';

function Home(props) {
  return (
    <View
      style={{
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
      }} >
      <TouchableOpacity
        style={{
          width: 200,
          height: 50,
          borderColor: 'black',
          borderWidth: 1,
          borderRadius: 5,
          justifyContent: 'center',
          alignItems: 'center'
        }}
        onPress={() => {
          props.navigator.push({
            screen: 'example.Screen2'
          });
        }} >
        <Text>
          Push
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={{
          width: 200,
          height: 50,
          borderColor: 'black',
          borderWidth: 1,
          borderRadius: 5,
          justifyContent: 'center',
          alignItems: 'center',
          marginTop: 20
        }}
        onPress={() => {
          props.navigator.resetTo({
            screen: 'example.Home'
          });
        }} >
        <Text>
          Reset
        </Text>
      </TouchableOpacity>
    </View>
  );
}

function Screen2(props) {
  return (
    <View
      style={{
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
      }} >
      <TouchableOpacity
        style={{
          width: 200,
          height: 50,
          borderColor: 'black',
          borderWidth: 1,
          borderRadius: 5,
          justifyContent: 'center',
          alignItems: 'center'
        }}
        onPress={() => {
          props.navigator.pop();
        }} >
        <Text>
          Back
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={{
          width: 200,
          height: 50,
          borderColor: 'black',
          borderWidth: 1,
          borderRadius: 5,
          justifyContent: 'center',
          alignItems: 'center',
          marginTop: 20
        }}
        onPress={() => {
          props.navigator.resetTo({
            screen: 'example.Home'
          });
        }} >
        <Text>
          Reset
        </Text>
      </TouchableOpacity>
    </View>
  )
}

Navigation.registerComponent('example.Home', () => Home);
Navigation.registerComponent('example.Screen2', () => Screen2);

Navigation.startSingleScreenApp({
  screen: {
    screen: 'example.Home',
  }
});
```

### CODE REVIEW

**Comment by thanhzusu** [comment]
Thank you so much. I faced the same issue.

File: ios/RCCNavigationController.m [hunk 1]
```diff
@@ -249,6 +249,8 @@ - (void)performAction:(NSString*)performAction actionParams:(NSDictionary*)actio
     }
     
     BOOL animated = actionParams[@"animated"] ? [actionParams[@"animated"] boolValue] : YES;
+
+    [self.viewControllers.lastObject viewWillDisappear:animated];
```
Comment (thanhzusu) [inline ios/RCCNavigationController.m]: For safety, could you validate size of self.viewControllers?

Comment (Rudolph-Miller) [reply ios/RCCNavigationController.m]: OK!

**Comment by Rudolph-Miller** [comment]
@thanhzusu

**Comment by rgoldiez** [comment]
Any reason this isn't getting merged?  It's a straightforward fix and works seemingly without adverse side effects.

**Comment by Rudolph-Miller** [comment]
https://github.com/drpiou/react-native-navigation/commit/522e6e9753c26d8be73ac6828dbcfc4209e97e94 fixed the issue.

**Comment by Rudolph-Miller** [comment]
Oh, I see.
https://github.com/drpiou/react-native-navigation/commit/522e6e9753c26d8be73ac6828dbcfc4209e97e94 is just a commit to forked REPO.

**Pull request closed without merging**